Mathlib/Tactic/Linter/Style.lean

```diff
@@ -466,4 +468,50 @@ initialize addLinter doubleUnderscore
 
 end Style.nameCheck
 
+/-! # The "openClassical" linter -/
+
+/-- The "openClassical" linter emits a warning on `open Classical` statements which are not
+scoped to a single declaration. A non-scoped `open Classical` can hide that some theorem statements
+would be better stated with explicit decidability statements.
+-/
+register_option linter.style.openClassical : Bool := {
+  defValue := true
+  descr := "enable the openClassical linter"
+}
+
+namespace Style.openClassical
+
+/-- If `stx` is syntax describing an `open` command, `extractOpenNames stx`
+returns an array of the syntax corresponding to the opened names,
+omitting any renamed or hidden items. -/
+def extractOpenNames : Syntax → Array Syntax
```

Comment:
I remember this function already existing, but I cannot find it.  Do you know if it already exists and if maybe it can be recycled?

Reply:
I don't remember seeing it, but have [asked on zulip](https://leanprover.zulipchat.com/#narrow/channel/217875-Is-there-code-for-X.3F/topic/Find.20open.20statements.20in.20syntax)

Reply:
Status update: nobody responded on zulip --- perhaps it doesn't exist yet.

Reply:
Ok, I must me misremembering!